### ISSUE TITLE
Highlighter produces ill-formed output given legacy Mac OS line endings (LF)

### ISSUE THREAD

**Issue by patrickt** [tree-sitter/tree-sitter#2344]
You can find such a file here, in `diffusionclip.py`: https://github.com/gwang-kim/DiffusionCLIP

The CLI output only renders a couple random lines from the file. The HTML output is garbled.

**Comment by ahelwer** [comment]
Does it even parse correctly from the CLI? If not, this is probably an issue with tree-sitter-python if they want to support those line endings.

**Comment by patrickt** [comment]
@ahelwer Yes, the parsing (at least with tree-sitter-python) seems fine via `tree-sitter parse`. The lexer for Python appears to handle LFs correctly, so I think the bug is somewhere in the highlighter. And yes, the fact that that file appears on one line is a direct result of this bug 😄

**Comment by amaanq** [comment]
hmm - highlights look fine, is this not ok?

![image](https://github.com/tree-sitter/tree-sitter/assets/29718261/a7ccdf46-998c-4471-b653-c9c59ed637dd)

**Comment by amaanq** [comment]
Closing since this is just ancient and not worth supporting today

**Comment by patrickt** [comment]
@amaanq I’d love it if we could take another look at supporting this—there’s a lot of legacy code on GitHub, and users are hitting display bugs because of it.

**Comment by sperlis** [comment]
This is a problem that keeps recurring. I just faced that several times in the last few weeks. Since this bug is still marked open, does that mean it's still being considered. Is it possible to prioritize or something?
